fuel/streams.py

```diff
@@ -101,8 +125,18 @@ class DataStream(AbstractDataStream):
 
     """
     def __init__(self, dataset, **kwargs):
-        kwargs.setdefault('axis_labels', dataset.axis_labels)
+        if dataset.axis_labels:
+            kwargs.setdefault('axis_labels', dataset.axis_labels.copy())
         super(DataStream, self).__init__(**kwargs)
+        # A DataStream with no iteration scheme is considered an example stream
+        # by default
+        if not self.iteration_scheme:
```

Comment:
Is there ever a case where it would make sense for a `DataStream` to have `produces_examples` set to `False` as a kwarg?